### ISSUE TITLE
Qwen2.5VL-7B 32B本地推理回复乱码

### ISSUE THREAD

**Issue by ShbGao-ProMax** [QwenLM/Qwen2.5-VL#1343]
Qwen2.5VL-7B 32B本地推理回复乱码 

目前的实验结果是 使用VLLM进行推理完全没问题 可以确定模型权重是没问题的 我代码中的测试解码编码也没问题 但是使用transformers 和 modelscope进行推理都是回复乱码 

且乱码生成的非常稳定 请问该如何解决？

源码：

`from modelscope import Qwen2_5_VLForConditionalGeneration, AutoProcessor
from qwen_vl_utils import process_vision_info
import torch

from transformers import AutoTokenizer

tokenizer = AutoTokenizer.from_pretrained("/mnt/ShareDB_6TB/shibogao/checkpoint/Qwen2.5-VL-7B-Instruct-modelscope")
test_text = "请描述这个菜，给出评价"
tokens = tokenizer(test_text, return_tensors="pt")
print(tokens)
decoded_text = tokenizer.decode(tokens['input_ids'][0])
print(decoded_text)  # 应该返回原始文本

# 加载模型和分词器
model = Qwen2_5_VLForConditionalGeneration.from_pretrained(
    "/mnt/ShareDB_6TB/shibogao/checkpoint/Qwen2.5-VL-7B-Instruct-modelscope", 
    torch_dtype="auto", 
    device_map="auto"
)
processor = AutoProcessor.from_pretrained(
    "/mnt/ShareDB_6TB/shibogao/checkpoint/Qwen2.5-VL-7B-Instruct-modelscope"
)

# 定义测试消息
messages = [
    {
        "role": "user",
        "content": [{"type": "text", "text": "请描述这个菜，给出评价"}]
    }
]

# 准备推理输入
text = processor.apply_chat_template(messages, tokenize=False, add_generation_prompt=True)
image_inputs, video_inputs = process_vision_info(messages)
inputs = processor(
    text=[text],
    images=image_inputs,
    videos=video_inputs,
    padding=True,
    return_tensors="pt",
)
inputs = inputs.to("cuda")  # 如果 GPU 不可用，可以替换为 "cpu"

# 推理
generated_ids = model.generate(**inputs, max_new_tokens=128)
generated_ids_trimmed = [
    out_ids[len(in_ids) :] for in_ids, out_ids in zip(inputs.input_ids, generated_ids)
]
output_text = processor.batch_decode(
    generated_ids_trimmed, skip_special_tokens=True, clean_up_tokenization_spaces=True
)

print("Generated IDs:", generated_ids)

# 输出结果
print('------------------------------------------begin------------------------------------------')
for line in output_text:
    print(line.encode('utf-8').decode('utf-8'))  # 确保输出处理正确
print('------------------------------------------end------------------------------------------')

# 打印分词器的词汇表
tokenizer = processor.tokenizer
vocab = tokenizer.get_vocab()
id_to_token = {v: k for k, v in vocab.items()}  # ID -> Token 映射

# 打印部分词汇表
print("Vocabulary size:", len(vocab))
print("Example tokens:", list(vocab.items())[:10])

# 打印每个 ID 对应的 Token
generated_ids_list = generated_ids[0].tolist()  # 获取第一个句子的 ID 列表
decoded_tokens = [id_to_token.get(token_id, "<UNK>") for token_id in generated_ids_list]

print("Generated IDs:", generated_ids_list)
print("Decoded Tokens:", decoded_tokens)

# 手动解码每个 ID
for token_id in generated_ids_list:
    token = tokenizer.decode([token_id], skip_special_tokens=False)
    print(f"ID: {token_id}, Token: {token}")

# 手动解码完整句子
decoded_text = tokenizer.decode(generated_ids_list, skip_special_tokens=True)
print("Decoded Text:", decoded_text)`

输出 

`{'input_ids': tensor([[ 14880,  53481,  99487,  99800,   3837, 107485, 103964]]), 'attention_mask': tensor([[1, 1, 1, 1, 1, 1, 1]])}
请描述这个菜，给出评价
Loading checkpoint shards: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 5/5 [00:05<00:00,  1.18s/it]
Using a slow image processor as `use_fast` is unset and a slow processor was saved with this model. `use_fast=True` will be the default behavior in v4.52, even if the model was saved with a slow processor. This will result in minor differences in outputs. You'll still be able to use a slow processor with `use_fast=False`.
You have video processor config saved in `preprocessor.json` file which is deprecated. Video processor configs should be saved in their own `video_preprocessor.json` file. You can rename the file or load and save the processor back which renames it automatically. Loading from `preprocessor.json` will be removed in v5.0.
Generated IDs: tensor([[151644,   8948,    198,   2610,    525,    264,  10950,  17847,     13,
         151645,    198, 151644,    872,    198,  14880,  53481,  99487,  99800,
           3837, 107485, 103964, 151645,    198, 151644,  77091,    198,      7,
             13,   5512,  23620,   1528,     16,  79529,    481,   6164,     87,
           9866,  11959,    465,     84,   3251,     16,   1143,  61217,     16,
           2863,  91777, 116948,     22, 101545,    277,     13,   5512,     46,
             12,  13524,     13,   5512,   5338,   1528,  80278,    220,     11,
           9748,   5338,   1528,     38,     78,     11,     64,    422,   2082,
          12907,      1,   5248,    366,    320,   9748,   5338,   4586,     13,
           5512, 104951, 100070,     13,  41778,     13,   5512,    303,    856,
            485,     78,     11,   2577,   2073,  48235,     13,   5338,  47821,
            267,   5338, 134970,    320,    220,     11,   7977, 116820,     11,
           2577,    320,     12,     13,   7135,   2876,     79,     11,   5235,
           2577,    320,    220,     11,    856, 104015,   3533,    320,    220,
             11,   7977,   7135,   2876,    277, 104015,  12676,     47,  11426,
            320,    220,    320,    220,    320,    220,    320,    220,    320,
            220,    320,    220,    320,    220,    320,    220,    320,    220,
            320]], device='cuda:0')
------------------------------------------begin------------------------------------------
(. First-Oche1 Grape - Bigx Location Olteru General1Ch erste1au老器具7柴ar. FirstO-.Clear. FirstFirstcheacock, AffFirstcheGo,a DAn Af" multiple < ( AffFirst general. First秩序公共.GUID. Firstnd xereo,af“ Chin.FirstterrastFirstпат (, af提早,af (-. Pat Marp, Blackaf (, x迅速Pr (, af Pat Marar迅速PublicP Land (  (  (  (  (  (  (  (  (  (
------------------------------------------end------------------------------------------
Vocabulary size: 151665
Example tokens: [('ĠCPU', 13940), ('minor', 45670), ('ï¬', 123954), ('Fetcher', 97492), ('Traffic', 87229), ('enef', 50465), ('ĠÐ½Ð¸Ñħ', 129690), ('.swift', 13290), ('Ġdull', 40530), ('ï³Ĳ', 148682)]
Generated IDs: [151644, 8948, 198, 2610, 525, 264, 10950, 17847, 13, 151645, 198, 151644, 872, 198, 14880, 53481, 99487, 99800, 3837, 107485, 103964, 151645, 198, 151644, 77091, 198, 7, 13, 5512, 23620, 1528, 16, 79529, 481, 6164, 87, 9866, 11959, 465, 84, 3251, 16, 1143, 61217, 16, 2863, 91777, 116948, 22, 101545, 277, 13, 5512, 46, 12, 13524, 13, 5512, 5338, 1528, 80278, 220, 11, 9748, 5338, 1528, 38, 78, 11, 64, 422, 2082, 12907, 1, 5248, 366, 320, 9748, 5338, 4586, 13, 5512, 104951, 100070, 13, 41778, 13, 5512, 303, 856, 485, 78, 11, 2577, 2073, 48235, 13, 5338, 47821, 267, 5338, 134970, 320, 220, 11, 7977, 116820, 11, 2577, 320, 12, 13, 7135, 2876, 79, 11, 5235, 2577, 320, 220, 11, 856, 104015, 3533, 320, 220, 11, 7977, 7135, 2876, 277, 104015, 12676, 47, 11426, 320, 220, 320, 220, 320, 220, 320, 220, 320, 220, 320, 220, 320, 220, 320, 220, 320, 220, 320]
Decoded Tokens: ['<|im_start|>', 'system', 'Ċ', 'You', 'Ġare', 'Ġa', 'Ġhelpful', 'Ġassistant', '.', '<|im_end|>', 'Ċ', '<|im_start|>', 'user', 'Ċ', 'è¯·', 'æııè¿°', 'è¿Ļä¸ª', 'èıľ', 'ï¼Į', 'ç»Ļåĩº', 'è¯Ħä»·', '<|im_end|>', 'Ċ', '<|im_start|>', 'assistant', 'Ċ', '(', '.', 'ĠFirst', '-O', 'che', '1', 'ĠGrape', 'Ġ-', 'ĠBig', 'x', 'ĠLocation', 'ĠOl', 'ter', 'u', 'ĠGeneral', '1', 'Ch', 'Ġerste', '1', 'au', 'èĢģ', 'åĻ¨åħ·', '7', 'æŁ´', 'ar', '.', 'ĠFirst', 'O', '-', '.Clear', '.', 'ĠFirst', 'First', 'che', 'acock', 'Ġ', ',', 'ĠAff', 'First', 'che', 'G', 'o', ',', 'a', 'ĠD', 'An', 'ĠAf', '"', 'Ġmultiple', 'Ġ<', 'Ġ(', 'ĠAff', 'First', 'Ġgeneral', '.', 'ĠFirst', 'ç§©åºı', 'åħ¬åħ±', '.', 'GUID', '.', 'ĠFirst', 'nd', 'Ġx', 'ere', 'o', ',', 'af', 'âĢľ', 'ĠChin', '.', 'First', 'terra', 'st', 'First', 'Ð¿Ð°ÑĤ', 'Ġ(', 'Ġ', ',', 'Ġaf', 'æıĲæĹ©', ',', 'af', 'Ġ(', '-', '.', 'ĠPat', 'ĠMar', 'p', ',', 'ĠBlack', 'af', 'Ġ(', 'Ġ', ',', 'Ġx', 'è¿ħéĢŁ', 'Pr', 'Ġ(', 'Ġ', ',', 'Ġaf', 'ĠPat', 'ĠMar', 'ar', 'è¿ħéĢŁ', 'Public', 'P', 'ĠLand', 'Ġ(', 'Ġ', 'Ġ(', 'Ġ', 'Ġ(', 'Ġ', 'Ġ(', 'Ġ', 'Ġ(', 'Ġ', 'Ġ(', 'Ġ', 'Ġ(', 'Ġ', 'Ġ(', 'Ġ', 'Ġ(', 'Ġ', 'Ġ(']
ID: 151644, Token: <|im_start|>
ID: 8948, Token: system
ID: 198, Token: 

ID: 2610, Token: You
ID: 525, Token:  are
ID: 264, Token:  a
ID: 10950, Token:  helpful
ID: 17847, Token:  assistant
ID: 13, Token: .
ID: 151645, Token: <|im_end|>
ID: 198, Token: 

ID: 151644, Token: <|im_start|>
ID: 872, Token: user
ID: 198, Token: 

ID: 14880, Token: 请
ID: 53481, Token: 描述
ID: 99487, Token: 这个
ID: 99800, Token: 菜
ID: 3837, Token: ，
ID: 107485, Token: 给出
ID: 103964, Token: 评价
ID: 151645, Token: <|im_end|>
ID: 198, Token: 

ID: 151644, Token: <|im_start|>
ID: 77091, Token: assistant
ID: 198, Token: 

ID: 7, Token: (
ID: 13, Token: .
ID: 5512, Token:  First
ID: 23620, Token: -O
ID: 1528, Token: che
ID: 16, Token: 1
ID: 79529, Token:  Grape
ID: 481, Token:  -
ID: 6164, Token:  Big
ID: 87, Token: x
ID: 9866, Token:  Location
ID: 11959, Token:  Ol
ID: 465, Token: ter
ID: 84, Token: u
ID: 3251, Token:  General
ID: 16, Token: 1
ID: 1143, Token: Ch
ID: 61217, Token:  erste
ID: 16, Token: 1
ID: 2863, Token: au
ID: 91777, Token: 老
ID: 116948, Token: 器具
ID: 22, Token: 7
ID: 101545, Token: 柴
ID: 277, Token: ar
ID: 13, Token: .
ID: 5512, Token:  First
ID: 46, Token: O
ID: 12, Token: -
ID: 13524, Token: .Clear
ID: 13, Token: .
ID: 5512, Token:  First
ID: 5338, Token: First
ID: 1528, Token: che
ID: 80278, Token: acock
ID: 220, Token:  
ID: 11, Token: ,
ID: 9748, Token:  Aff
ID: 5338, Token: First
ID: 1528, Token: che
ID: 38, Token: G
ID: 78, Token: o
ID: 11, Token: ,
ID: 64, Token: a
ID: 422, Token:  D
ID: 2082, Token: An
ID: 12907, Token:  Af
ID: 1, Token: "
ID: 5248, Token:  multiple
ID: 366, Token:  <
ID: 320, Token:  (
ID: 9748, Token:  Aff
ID: 5338, Token: First
ID: 4586, Token:  general
ID: 13, Token: .
ID: 5512, Token:  First
ID: 104951, Token: 秩序
ID: 100070, Token: 公共
ID: 13, Token: .
ID: 41778, Token: GUID
ID: 13, Token: .
ID: 5512, Token:  First
ID: 303, Token: nd
ID: 856, Token:  x
ID: 485, Token: ere
ID: 78, Token: o
ID: 11, Token: ,
ID: 2577, Token: af
ID: 2073, Token: “
ID: 48235, Token:  Chin
ID: 13, Token: .
ID: 5338, Token: First
ID: 47821, Token: terra
ID: 267, Token: st
ID: 5338, Token: First
ID: 134970, Token: пат
ID: 320, Token:  (
ID: 220, Token:  
ID: 11, Token: ,
ID: 7977, Token:  af
ID: 116820, Token: 提早
ID: 11, Token: ,
ID: 2577, Token: af
ID: 320, Token:  (
ID: 12, Token: -
ID: 13, Token: .
ID: 7135, Token:  Pat
ID: 2876, Token:  Mar
ID: 79, Token: p
ID: 11, Token: ,
ID: 5235, Token:  Black
ID: 2577, Token: af
ID: 320, Token:  (
ID: 220, Token:  
ID: 11, Token: ,
ID: 856, Token:  x
ID: 104015, Token: 迅速
ID: 3533, Token: Pr
ID: 320, Token:  (
ID: 220, Token:  
ID: 11, Token: ,
ID: 7977, Token:  af
ID: 7135, Token:  Pat
ID: 2876, Token:  Mar
ID: 277, Token: ar
ID: 104015, Token: 迅速
ID: 12676, Token: Public
ID: 47, Token: P
ID: 11426, Token:  Land
ID: 320, Token:  (
ID: 220, Token:  
ID: 320, Token:  (
ID: 220, Token:  
ID: 320, Token:  (
ID: 220, Token:  
ID: 320, Token:  (
ID: 220, Token:  
ID: 320, Token:  (
ID: 220, Token:  
ID: 320, Token:  (
ID: 220, Token:  
ID: 320, Token:  (
ID: 220, Token:  
ID: 320, Token:  (
ID: 220, Token:  
ID: 320, Token:  (
ID: 220, Token:  
ID: 320, Token:  (
Decoded Text: system
You are a helpful assistant.
user
请描述这个菜，给出评价
assistant
(. First-Oche1 Grape - Bigx Location Olteru General1Ch erste1au老器具7柴ar. FirstO-.Clear. FirstFirstcheacock , AffFirstcheGo,a DAn Af" multiple < ( AffFirst general. First秩序公共.GUID. Firstnd xereo,af“ Chin.FirstterrastFirstпат ( , af提早,af (-. Pat Marp, Blackaf ( , x迅速Pr ( , af Pat Marar迅速PublicP Land (  (  (  (  (  (  (  (  (  (`

我尝试了 视频输入 图像输入 和纯文本输入 都是乱码回复

**Comment by ShbGao-ProMax** [comment]
@JJJYmmm 请问您是否可以提供协助？

**Comment by JJJYmmm** [comment]
transformers具体版本是哪个

**Comment by ShbGao-ProMax** [comment]
> transformers具体版本是哪个

谢谢您的协助 经排查 我发现在其中一台服务器会出现此问题 其它几台服务器都非常正常 我们组内人初步确定应该是该服务器CUDA有部分损坏或安装不成功 您提供的inference sample没问题 再次感谢您的及时支持

**Comment by imzhangyd** [comment]
@JJJYmmm 请问可以帮看下这个问题吗[https://github.com/QwenLM/Qwen2.5-VL/issues/1445#issue-3372615819](https://github.com/QwenLM/Qwen2.5-VL/issues/1445#issue-3372615819)
